### PR TITLE
add batch job

### PR DESCRIPTION
**Pull request by MichaelJacob21012**
### Jira link
https://tools.hmcts.net/jira/browse/EM-6247

### Change description
add batch job to update missing jurisdiction codes. This feature/job is disabled via feature flag. We are awaiting the actual data/values from business team before running this job.

### CODE REVIEW

**Review by yogesh-hullatti — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/em/hrs/batch/UpdateJurisdictionCodesConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                      .build()).build().build();
+    }
+
+    private JpaPagingItemReader<HearingRecordingSegment> undeletedDocumentsWithTtl() {
+        return new JpaPagingItemReaderBuilder<HearingRecordingSegment>()
+            .name("documentTaskReader")
+            .entityManagerFactory(entityManagerFactory)
+            .queryString("select hrs from HearingRecordingSegment hrs where hrs.hearingRecording in "
+                             + "(select hr from HearingRecording hr where hr.serviceCode is null and "
+                             + "hr.jurisdictionCode is null order by createdOn asc)")
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/batch/UpdateJurisdictionCodesConfiguration.java:107]: is it ok to run this without limit?

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -112,15 +120,18 @@ azure:
...
       cvp-dest-blob-container-name: ${HRS_CVP_DEST_CONTAINER_NAME:emhrstestcontainer}
       vh-dest-blob-container-name: ${HRS_VH_DEST_CONTAINER_NAME:emhrsvhtestcontainer}
     cvp:
-      connection-string: ${CVP_STORAGE_CONNECTION_STRING:UseDevelopmentStorage=true}
+      connection-string: ${CVP_STORAGE_CONNECTION_STRING:DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://localhost:10000/devstoreaccount1}
       blob-container-reference: ${CVP_STORAGE_CONTAINER_NAME:cvptestcontainer}
     vh:
-      connection-string: ${VH_STORAGE_CONNECTION_STRING:UseDevelopmentStorage=true}
+      connection-string: ${VH_STORAGE_CONNECTION_STRING:DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://localhost:10000/devstoreaccount1}
```
Comment (mdayican) [inline src/main/resources/application.yaml:130]: why did we remove UseDevelopmentStorage

Comment (MichaelJacob21012) [reply src/main/resources/application.yaml:130]: Local configuration is only working with the long connection string

File: src/main/java/uk/gov/hmcts/reform/em/hrs/batch/UpdateJurisdictionCodesConfiguration.java (hunk 1 above)
Comment (MichaelJacob21012) [reply src/main/java/uk/gov/hmcts/reform/em/hrs/batch/UpdateJurisdictionCodesConfiguration.java:107]: Limit is not supported in HQL. We are using maxItemCount instead.

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/em/hrs/batch/UpdateJurisdictionCodesConfiguration.java:107]: I meant for inner sql

File: src/main/java/uk/gov/hmcts/reform/em/hrs/job/UpdateJurisdictionCodesTask.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                ));
+
+                if (records.size() >= batchSize) {
+                    batchUpdate(records);
+                    records.clear();
+                }
+            }
+
+            if (!records.isEmpty()) {
+                batchUpdate(records);
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/job/UpdateJurisdictionCodesTask.java:78]: what is the max for records?

**Review by mdayican — DISMISSED** [review]

**Review by yogesh-hullatti — APPROVED** [review]

**Review by mdayican — APPROVED** [review]

**Review by Raihan-R — APPROVED** [review]

**Pull request merged**